toolchain/c-toolchain.lisp

```diff
@@ -311,7 +311,10 @@ is bound to a temporary file name, then atomically renaming that temporary file
   #-(or ecl mkcl)
   ;; Don't use a temporary file, because linking is sensitive to the output file name :-/ (or put it in a temporary directory?)
   (apply 'invoke *ld* "-o" output-file
-         (append *ld-dll-flags* inputs)))
+         (append *ld-dll-flags*
+		 #-(or windows darwin) (list (format nil "-Wl,-soname=~a"
+						     (file-namestring output-file)))
+		 inputs)))
```

Comment:
I think you can drop this too. When we set up the proper hooks for unloading/reloading the library, the soname shouldn't matter. (And I thought `-Wl` was a gcc option, not an ld option.)

Reply:
-Wl is a gcc option, but it just sends the -soname option to the linker.  I think cffi still uses gcc to invoke the linker so *ld-dll-flags* is a bit of a misnomer.  For some reason I think I remember trying the image hooks first without the soname and running into a surprise on one of ccl or sbcl (but maybe it was vice versa).   I'll test to confirm its unnecessary on a scratch tree with proper close/use hooks, but yeah, simpler compiler flags would be better even if soname's are used on the majority of linux C foreign library builds.